### PR TITLE
Mb share jpa date tests

### PR DESCRIPTION
**Pull request by michaelabuckley**
Share the new date tests with Mongo.

### CODE REVIEW

**Review by katiesmilecdr — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/search/ExtendedLuceneIndexExtractor.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public ExtendedLuceneIndexExtractor(FhirContext theContext, Map<String, RuntimeS
 
 	@NotNull
 	public ExtendedLuceneIndexData extract(ResourceIndexedSearchParams theNewParams) {
-		// wip mb this is testable now.
+		// wipmb this is testable now.
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/search/ExtendedLuceneIndexExtractor.java:49]: is this needed still?

File: hapi-fhir-storage/pom.xml [hunk 2]
```diff
@@ -151,6 +158,18 @@
...
 		</pluginManagement>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<version>3.2.0</version>
+				<executions>
+					<execution>
+						<goals>
+							<goal>test-jar</goal>
```
Comment (tadgh) [inline hapi-fhir-storage/pom.xml:168]: what is this doing exactly?

File: pom.xml [hunk 3]
```diff
@@ -2913,7 +2913,7 @@
 		<profile>
 			<id>FASTINSTALL</id>
 			<properties>
-				<maven.test.skip>true</maven.test.skip>
+				<skipTests>true</skipTests>
```
Comment (tadgh) [inline pom.xml:2916]: just verifying you tested this and it works.

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-storage/pom.xml (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-storage/pom.xml:168]: Builds a jar of all the test classes and publishes it as a test-jar type jar.
The alternative is another full project called something like hapi-fhir-storage-test-utils, but that seemed overkill for a couple of classes.

**Pull request merged**